contracts/model/DualGovernance.sol

```diff
@@ -0,0 +1,301 @@
+    uint256 public constant DYNAMIC_TIMELOCK_MIN_DURATION = 5 days; // L_min; minimum duration for the dynamic timelock, which extends based on the level of dissent or rage quit support.
+    uint256 public constant DYNAMIC_TIMELOCK_MAX_DURATION = 45 days; // L_max; maximum possible duration for dynamic timelocks, applied under conditions of extreme dissent to delay proposal execution.
+    uint256 public constant VETO_SIGNALLING_MIN_ACTIVE_DURATION = 5 hours; // Minimum time Veto Signalling must be active before before transitioning to the Deactivation sub-state can be considered.
+    uint256 public constant VETO_COOLDOWN_DURATION = 5 hours; // Cooling period following the Veto Signalling state to prevent immediate re-signalling.
+    uint256 public constant VETO_SIGNALLING_DEACTIVATION_MAX_DURATION = 3 days; // Maximum duration that the Veto Signalling can remain in Deactivation before advancing to Veto Cooldown or reverting to Veto Signalling.
+    uint256 public constant RAGE_QUIT_EXTENSION_DELAY = 7 days; // The delay follows the completion of the withdrawal process in Rage Quit state.
+
+    // Constructor to initialize the governance contract in the Normal state.
+    constructor(address _fakeETH, uint256 emergencyProtectionTimelock) {
+        currentState = State.Normal;
+        lastStateChangeTime = block.timestamp;
+        fakeETH = _fakeETH;
+        emergencyProtectedTimelock = new EmergencyProtectedTimelock(address(this), emergencyProtectionTimelock);
+        signallingEscrow = new Escrow(address(this), _fakeETH);
+    }
+
+    // Operations
+    /**
+     * Submits a proposal for consideration within the governance model.
+     * Proposals can be submitted when in the Normal state or during Veto Signalling; however they cannot be executed in Veto Signalling.
+     */
+    function submitProposal(ExecutorCall[] calldata calls) external returns (uint256 proposalId) {
+        activateNextState();
+
+        require(proposers[msg.sender], "Caller is not authorized to submit proposals.");
+        require(calls.length != 0, "Empty calls.");
+        require(
+            currentState == State.Normal || currentState == State.VetoSignalling || currentState == State.RageQuit,
+            "Cannot submit in current state."
+        );
+
+        proposalId = emergencyProtectedTimelock.submit(msg.sender, calls);
+    }
+
+    /**
+     * Schedules a proposal for execution, ensuring that all conditions for governance are met.
+     * Scheduling is allowed in Normal and Veto Cooldown states to prepare proposals for decision-making.
+     */
+    function scheduleProposal(uint256 proposalId) external {
+        activateNextState();
+
+        require(
+            currentState == State.Normal || currentState == State.VetoCooldown,
+            "Proposals can only be scheduled in Normal or Veto Cooldown states."
+        );
+        if (currentState == State.VetoCooldown) {
+            require(
```

Comment:
Correct me if I'm wrong but looks like it doesn't work in a correct way. 
State transition from normal to vetosignalling sets `lastVetoSignallingTime` to this moment so after any further state transition `lastVetoSignallingTime` will be always less that `block.timestamp`. So it's impossible to schedule proposal on vetocooldown state.